File: src/main/java/dk/frankbille/scoreboard/domain/League.java
package dk.frankbille.scoreboard.domain;

import java.io.Serializable;

import org.apache.commons.lang.builder.EqualsBuilder;
import org.apache.commons.lang.builder.HashCodeBuilder;

public class League implements Serializable {
	private static final long serialVersionUID = 1L;

	private Long id;
	
	private String name;

	public Long getId() {
		return id;
	}

	public void setId(Long id) {
		this.id = id;
	}

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}
	
	@Override
	public boolean equals(Object obj) {
		return EqualsBuilder.reflectionEquals(this, obj);
	}
	
	@Override
	public int hashCode() {
		return HashCodeBuilder.reflectionHashCode(this);
	}
	
}

File: restapi/league.go
package scoreboard

import (
    "github.com/frankbille/sanitize"
)

type League struct {
    Id     string `json:"id"`
    Name   string `json:"name"`
    Active bool `json:"active"`
}

func NewLeague(Name string, Active bool) League {
    return League{
        Id:     sanitize.Path(Name),
        Name:   Name,
        Active: Active,
    }
}

func (l League) GetId() string {
	return l.Id
}
